docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -399,6 +410,65 @@ For example, if you work with Keycloak, you can use `keycloak.js` to authenticat
 </html>
 ----
 
+[[single-page-application-oidc-proxy]]
+==== Use OIDC proxy
+
+Instead of tying your SPA to a specific OIDC provider by depending on this OIDC provider's script, you may want to consider implementing an authorization code flow directly in the SPA.
+
+SPA redirects users to a configurable OIDC authorization endpoint, accepts the OIDC provider's redirect and acquires tokens by sending the authorization code to a standard OIDC token endpoint.
+
+Quarkus introduces an experimental `quarkus-oidc-proxy` OIDC proxy extension which provides an authorization code flow support for Quarkus OIDC `service` applications. It can help to encapsule all the OIDC provider specific configuration details.
+
+To use OIDC proxy, all you need to do is to add the `quarkus-oidc-proxy` dependency to your Quarkus OIDC `service` application:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-oidc-proxy</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-oidc-proxy")
+----
+
+Now, the SPA can use the OIDC proxy endpoints to authenticate users with an authorization code flow and send the acquired access tokens to the Quarkus OIDC `service` application.
+
+For example, if the endpoint is listening at `http://localhost:8080`, then, after adding this dependency, the OIDC authorization endpoint is available at `http://localhost:8080/oidc/authorize` , the OIDC token endpoint is available at `http://localhost:8080/oidc/token`, the OIDC JWKS endpoint is available at `http://localhost:8080/oidc/jwks` and finally, the OIDC well known configuration endpoint which supports the discovery is available at `http://localhost:8080/oidc/.well-known/openid-configuration`. These endpoint paths can be customized.
```

Comment:
Are you sure about `http://localhost:8080/oidc/.well-known/openid-configuration`? This won't be usable for most users, no? It's not the standard path.

Reply:
@FroMage OIDC proxy acts as an OIDC provider itself to its clients, the configurable base path is `/oidc` (or whatever we will agree to), and a `/.well-known/openid-configuration` is what OIDC providers have for the clients who are capable of discovering the metadata to discover them. So this is fine. FYI, the test `web-app` tenant depends on this discovery feature

Reply:
@FroMage Hi Steph, however, if this proxy interposes over for example GitHub which has neither discovery not jwks endpoint then it is indeed unnecessary, so I'll make the addition of these 2 endpoints optional

Reply:
@FroMage I've got confused here, this `/.well-known/openid-configuration` represents the proxy's endpoints, authorization, token URLs, but indeed, whether the proxy JWKS endpoint is included in it or not will depend if the udnerlying provider can provide the keys

Reply:
@FroMage JWKS endpoint is already optional

Reply:
I'm not sure I understand what you explained. I am pretty sure that if we make the path `http://localhost:8080/oidc/.well-known/openid-configuration` (with a prefix) or anything else BUT `http://localhost:8080/.well-known/openid-configuration` (the standard) then clients won't be able to use it unless they have special configuration, no?

Reply:
@FroMage There is no standard as to what the base path is. I have the discovery tested in this PR. One points Quarkus OIDC or any other OIDC discovery aware client  to the OIDC service base path, in case of the OIDC proxy it ends with `/q/oidc` by default and appending `.well-known/openid-configuration` to it is what constitutes a standard/typical process.
For ex in Quarkus Renard one points to the Apple base path, etc, without explicitly appending `.well-known/openid-configuration` to it. So as far as OIDC proxy is concerned it is like Apple OIDC provider to the outside world, users decide what its base path is, `/q/oidc`, `/`, `/oidc` etc and that is it, the consumer points to it and the discovery is done. 

Does it clarify it ?

Reply:
Oh, OK, so you're saying we have to point to a provider URI, which optionally includes a path, and it doesn't need to be toplevel. We don't specify a provider host name or IP, which would be toplevel and not include a path. OK.

Thanks, this makes sense.

Reply:
@FroMage So, OIDC proxy base path the user decides upon is a standard base path as far as OIDC proxy emulating the OIDC provider is concerned...